frontend/src/app/south/edit-south-item-modal/edit-south-item-modal.component.ts

```diff
@@ -0,0 +1,143 @@
+import { Component } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { FormBuilder, Validators } from '@angular/forms';
+import { Observable, switchMap } from 'rxjs';
+import { ObservableState, SaveButtonComponent } from '../../shared/save-button/save-button.component';
+import { TranslateModule } from '@ngx-translate/core';
+import { formDirectives } from '../../shared/form-directives';
+import { SouthConnectorDTO, SouthItemCommandDTO, SouthItemDTO } from '../../model/south-connector.model';
+import { SouthConnectorService } from '../../services/south-connector.service';
+import { ScanModeDTO } from '../../model/scan-mode.model';
+import { NgForOf, NgIf, NgSwitch, NgSwitchCase } from '@angular/common';
+import { OibFormControl, OibScanModeFormControl } from '../../model/form.model';
+import { createInputs } from '../../shared/utils';
+import { OibScanModeComponent } from '../../shared/form/oib-scan-mode/oib-scan-mode.component';
+import { OibTextComponent } from '../../shared/form/oib-text/oib-text.component';
+import { OibTextAreaComponent } from '../../shared/form/oib-text-area/oib-text-area.component';
+import { OibCodeBlockComponent } from '../../shared/form/oib-code-block/oib-code-block.component';
+import { OibNumberComponent } from '../../shared/form/oib-number/oib-number.component';
+import { OibSelectComponent } from '../../shared/form/oib-select/oib-select.component';
+import { OibCheckboxComponent } from '../../shared/form/oib-checkbox/oib-checkbox.component';
+import { OibSecretComponent } from '../../shared/form/oib-secret/oib-secret.component';
+import { OibTimezoneComponent } from '../../shared/form/oib-timezone/oib-timezone.component';
+
+@Component({
+  selector: 'oib-edit-south-item-modal',
+  templateUrl: './edit-south-item-modal.component.html',
+  styleUrls: ['./edit-south-item-modal.component.scss'],
+  imports: [
+    ...formDirectives,
+    TranslateModule,
+    SaveButtonComponent,
+    NgForOf,
+    NgSwitch,
+    NgSwitchCase,
+    NgIf,
+    OibScanModeComponent,
+    OibTextComponent,
+    OibTextAreaComponent,
+    OibCodeBlockComponent,
+    OibNumberComponent,
+    OibSelectComponent,
+    OibCheckboxComponent,
+    OibSecretComponent,
+    OibTimezoneComponent
+  ],
+  standalone: true
+})
+export class EditSouthItemModalComponent {
+  mode: 'create' | 'edit' = 'create';
+  state = new ObservableState();
+  southConnector: SouthConnectorDTO | null = null;
+  southItemSchema: Array<OibFormControl> = [];
+  southItem: SouthItemDTO | null = null;
+  scanModes: Array<ScanModeDTO> = [];
+  scanModeOptions: OibScanModeFormControl | null = null;
+  form = this.fb.group({
+    name: ['', Validators.required],
+    scanMode: this.fb.group({ scanMode: [null as ScanModeDTO | null, Validators.required] }),
+    settings: this.fb.group({})
```

Comment:
I probably missed it in the previous PR, but settings should probably be a FormRecord (as I suppose you add keys dynamically)